src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java

```diff
@@ -64,14 +97,19 @@ protected int getDefaultPort() {
   }
 
   @Override
-  protected Map<String, String> createAuth() {
+  protected Map<String, String> createAuth(Map<String, Object> parameters) {
+    session = getSession();
     // Cassandra username are all lowercase and avoid numbers in the beginning.
     String username = generateRandomString(true).toLowerCase();
     String password = generateRandomString(false);
     String createRole = "CREATE ROLE " + username + " with superuser = false " +
         "and login = true and password = '" + password + "'";
+
     session.execute(createRole);
-    session.close();
+    String role = (String) parameters.getOrDefault("role", "admin");
```

Comment:
Do we need to check sanity of this `role` if it is not the default admin? As in that role exists etc..